searchlib/src/vespa/searchlib/attribute/imported_search_context.cpp

```diff
@@ -32,16 +47,127 @@ unsigned int ImportedSearchContext::approximateHits() const {
 
 std::unique_ptr<queryeval::SearchIterator>
 ImportedSearchContext::createIterator(fef::TermFieldMatchData* matchData, bool strict) {
+    if (_merger.hasArray()) {
+        if (_merger.emptyArray()) {
+            return SearchIterator::UP(new EmptySearch());
+        } else {
+            using Posting = btree::BTreeKeyData<uint32_t, int32_t>;
+            using DocIt = DocIdIterator<Posting>;
+            DocIt postings;
+            auto array = _merger.getArray();
+            postings.set(&array[0], &array[array.size()]);
+            return std::make_unique<AttributePostingListIteratorT<DocIt>>(true, matchData, postings);
+        }
+    }
     if (!strict) {
         return std::make_unique<AttributeIteratorT<ImportedSearchContext>>(*this, matchData);
     } else {
         return std::make_unique<AttributeIteratorStrict<ImportedSearchContext>>(*this, matchData);
     }
 }
 
+namespace {
+
+class WeightedRef {
+    EntryRef _revMapIdx;
+    int32_t  _weight;
+
+public:
+    WeightedRef(EntryRef revMapIdx, int32_t weight)
+        : _revMapIdx(revMapIdx),
+          _weight(weight)
+    {
+    }
+    EntryRef getRevMapIdx() const { return _revMapIdx; }
+    int32_t getWeight() const { return _weight; }
+};
+
+struct TargetResult {
```

Comment:
Why is this a struct when WeightedRef is not? Consider making WeightedRef a struct as well as they both only carry data.